### PR TITLE
 #359: Support objects with array syntax in Feedback

### PR DESCRIPTION
**Pull request by beiyi0207**
To be able to support object-backed array notation.

### CODE REVIEW

File: modules/View.js [hunk 1]
```diff
@@ -1479,8 +1479,12 @@
      * @return {Array<String>} The fully expanded subattribute names
      */
     __generateSubAttributes: function(attr, model) {
-      var i, attrName, remainder, subAttrs, values,
-        firstBracket = attr.indexOf('[]');
+      var attrName;
```
Comment (kentmw) [inline modules/View.js]: i feel like this took a backstep in readability

File: test/karma/viewFeedbackSpec.js [hunk 2]
```diff
@@ -122,4 +122,88 @@ describe('A View', function() {
...
+      expect(this.objectFeedbackView.change).toBe(1);
+      expect(this.objectFeedbackView.indexMap.bar).toBe('abc');
+      this.objectFeedbackView.resetIndexMap();
+      this.objectFeedbackView.$el.find('[data-model="foo[abc]"]').change();
+      expect(this.objectFeedbackView.change).toBe(2);
+      expect(this.objectFeedbackView.indexMap.bar).toBe('abc');
+      this.objectFeedbackView.resetIndexMap();
+      this.objectFeedbackView.$el.find('[data-model="foo[def]"]').change();
+      expect(this.objectFeedbackView.change).toBe(3);
+      expect(this.objectFeedbackView.indexMap.bar).toBe('def');
```
Comment (kentmw) [inline test/karma/viewFeedbackSpec.js]: neat!

File: modules/View.js (hunk 1 above)
Comment (beiyi0207) [reply modules/View.js]: Use
```
var remainder, subAttrs, values, indexes,
    firstBracket = attr.indexOf('[]');
```
you mean?

Comment (kentmw) [reply modules/View.js]: yea. admittedly, im very much in the mindset of es6 now that doesn't have to hoist. but, having them all on one line is my least favorite

**Pull request merged**